crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -201,6 +206,7 @@ pub enum ScopeKind {
     Class,
     Function,
     Comprehension,
+    TypeAlias,
```

Comment:
A `TypeAlias` scope should also be included in `ScopeId::is_function_like()` above (see https://github.com/python/cpython/blob/main/Python/symtable.c#L568 ).

Note though that a `TypeAliasTypeParameters` scope should just be `ScopeKind::Annotation` like the other type-param scopes; the inner scope for the RHS of the alias should be `ScopeKind::TypeAlias`.